lib/autoupdate/agent/setup.go

```diff
@@ -246,9 +269,26 @@ func (ns *Namespace) Teardown(ctx context.Context) error {
 	if err := svc.Sync(ctx); err != nil {
 		return trace.Wrap(err, "failed to sync systemd config")
 	}
-	if err := os.RemoveAll(ns.versionsDir); err != nil {
+	if err := os.RemoveAll(namespaceDir(ns.name)); err != nil {
 		return trace.Wrap(err, "failed to remove versions directory")
 	}
+	if ns.name == "" {
+		oldTimer := &SystemdService{
+			ServiceName: deprecatedTimerName,
+			Log:         ns.log,
+		}
+		// If the old upgrader exists, attempt to re-enable it automatically
+		present, err := oldTimer.IsPresent(ctx)
+		if err != nil {
+			return trace.Wrap(err, "failed to determine if deprecated teleport-upgrade systemd timer is present")
```

Comment:
Ditto for keeping going if this branch fails, as the comment above?

I suggest the same refactor too - pull the branch logic to a function, have it error normally and only log/swallow here.

Reply:
Similar to above, returning an error is okay here.